### PR TITLE
feat(Tracker.gg): add presence

### PR DESCRIPTION
**Pull request by OGMatrix**
Hello!
Another Presence from my side, this Presence is for the Website [Tracker.gg](https://tracker.gg) where gamers can see their Stats in many Games. Currently the Presence supports only the Homepage and the Valorant Game Pages as you can see in the Code. I will make many updates that will always add one more Game Page to the Presence. Later you can also choose wich you want to show and wich not in the Settings. So here are **some** pictures!

**Tracker.gg Homepage**
image

**Tracker.gg Developer Homepage**
image

**Tracker.gg Dev Documentation**
image

**Tracker.gg Dev APIs**
image

**Tracker.gg Valorant Profile Page**
image

**Watching Valorant Lineup Clip**
image

**Watching Agents**
image

This were all interesting screenshots!

Have a nice day and im happy to see your Feedback!

### CODE REVIEW

**Comment by OGMatrix** [comment]
Hello, is it okey when the Lint Check isn't successful? Thanks!

**Comment by EncryptedDev** [comment]
> Hello, is it okey when the Lint Check isn't successful? Thanks!

We wouldn’t have it if it was optional.

**Comment by OGMatrix** [comment]
> > Hello, is it okey when the Lint Check isn't successful? Thanks!
> 
> We wouldn’t have it if it was optional.

But I don't understand what the errors mean. He tells me to do a line break, but there is already one.

**Comment by OGMatrix** [comment]
Hello! I have a Question, how should I name the File when I have to URLs added? Thanks!

File: websites/T/Tracker.gg/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.6",
+  "author": {
+    "name": "OGMatrix#8818",
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/dist/metadata.json:4]: The author name does not include the discriminator
```suggestion
    "name": "OGMatrix",
```

File: websites/T/Tracker.gg/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "id": "657290880947978241"
+  },
+  "service": "Tracker.gg",
+  "description": {
+    "en": "Tracker Network provides stats, global and regional leaderboards and much more to gamers around the world.",
+    "de": "Tracker Network bietet Spielern auf der ganzen Welt Statistiken, globale und regionale Bestenlisten und vieles mehr."
+  },
+  "url": ["tracker.gg", "fortnitetracker.com"],
+  "version": "1.0.1",
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/dist/metadata.json:13]:
```suggestion
  "version": "1.0.0",
```

File: websites/T/Tracker.gg/presence.ts [hunk 3]
```diff
@@ -0,0 +1,410 @@
+const trackerPresence = new Presence({ clientId: "929349462365704222" }),
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:1]: use `presence` instead

File: websites/T/Tracker.gg/presence.ts [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+  trackerBrowsing = Math.floor(Date.now() / 1000);
+
+trackerPresence.on("UpdateData", async () => {
+  const trackerPreData: PresenceData = {
+      largeImageKey: "logo",
+      startTimestamp: trackerBrowsing
+    },
+    trackerPage = window.location.pathname,
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:9]:
```suggestion
    { pathname } = window.location,
```

File: websites/T/Tracker.gg/presence.ts [hunk 5]
```diff
@@ -0,0 +1,410 @@
+const trackerPresence = new Presence({ clientId: "929349462365704222" }),
+  trackerBrowsing = Math.floor(Date.now() / 1000);
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:2]:
```suggestion
  browsingTimestamp = Math.floor(Date.now() / 1000);
```

File: websites/T/Tracker.gg/presence.ts [hunk 6]
```diff
@@ -0,0 +1,410 @@
...
+    } else if (trackerPage === "/developers/apps/create") {
+      trackerPreData.details = "Creating";
+      trackerPreData.state = "Tracker Dev App";
+    } else if (trackerPage.includes("/developers/apps")) {
+      try {
+        const appName: string = document
+          .querySelector(
+            "#app > div.trn-wrapper > div.trn-container > div > main > div.apps > div.content-container > div > div > div > div > div.dashboard__analytics.card.bordered.responsive > header > div > div.left > div:nth-child(2) > div.title"
+          )
+          .textContent.replace("Dashboard", "");
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:31]:
```suggestion
        const appName = document
          .querySelector<HTMLDivElement>(
            "#app > div.trn-wrapper > div.trn-container > div > main > div.apps > div.content-container > div > div > div > div > div.dashboard__analytics.card.bordered.responsive > header > div > div.left > div:nth-child(2) > div.title"
          )
          .textContent.replace("Dashboard", "");
```

File: websites/T/Tracker.gg/presence.ts [hunk 7]
```diff
@@ -0,0 +1,410 @@
...
+        trackerPreData.details = "Editing App:";
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:32]:
```suggestion
        trackerPreData.details = "Editing App";
```

File: websites/T/Tracker.gg/presence.ts [hunk 8]
```diff
@@ -0,0 +1,410 @@
...
+        trackerPreData.state = appName;
+        trackerPreData.smallImageKey = document
+          .querySelector(
+            "#app > div.trn-wrapper > div.trn-container > div > main > div.apps > div.content-container > div > div > div > div > div.dashboard__details.card.responsive > div > div:nth-child(1) > div > img"
+          )
+          .getAttribute("src")
+          .toString();
+        trackerPreData.smallImageText = appName;
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:40]: Not needed

File: websites/T/Tracker.gg/presence.ts [hunk 9]
```diff
@@ -0,0 +1,410 @@
...
+          )
+          .textContent.replace("Dashboard", "");
+        trackerPreData.details = "Editing App:";
+        trackerPreData.state = appName;
+        trackerPreData.smallImageKey = document
+          .querySelector(
+            "#app > div.trn-wrapper > div.trn-container > div > main > div.apps > div.content-container > div > div > div > div > div.dashboard__details.card.responsive > div > div:nth-child(1) > div > img"
+          )
+          .getAttribute("src")
+          .toString();
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:39]: How can't it be a string prior to the string conversion?

File: websites/T/Tracker.gg/presence.ts [hunk 10]
```diff
@@ -0,0 +1,410 @@
...
+      try {
+        const appName: string = document
+          .querySelector(
+            "#app > div.trn-wrapper > div.trn-container > div > main > div.apps > div.content-container > div > div > div > div > div.dashboard__analytics.card.bordered.responsive > header > div > div.left > div:nth-child(2) > div.title"
+          )
+          .textContent.replace("Dashboard", "");
+        trackerPreData.details = "Editing App:";
+        trackerPreData.state = appName;
+        trackerPreData.smallImageKey = document
+          .querySelector(
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:35]:
```suggestion
          .querySelector<HTMLImageElement>(
```

File: websites/T/Tracker.gg/presence.ts [hunk 11]
```diff
@@ -0,0 +1,410 @@
...
+            "#app > div.trn-wrapper > div.trn-container > div > main > div.apps > div.content-container > div > div > div > div > div.dashboard__details.card.responsive > div > div:nth-child(1) > div > img"
+          )
+          .getAttribute("src")
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:38]: Could just use the `.src` property

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/T/Tracker.gg/presence.ts (hunk 11 above)
Comment (OGMatrix) [reply websites/T/Tracker.gg/presence.ts]: Fixed!

File: websites/T/Tracker.gg/presence.ts [hunk 12]
```diff
@@ -0,0 +1,407 @@
+const presence = new Presence({ clientId: "929349462365704222" }),
+  browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+  const trackerPreData: PresenceData = {
```
Comment (ImRodry) [inline websites/T/Tracker.gg/presence.ts:5]: Consistency
```suggestion
  const presenceData: PresenceData = {
```

**Review by ImRodry — CHANGES_REQUESTED** [review]
Also title does not follow commitlint

File: websites/T/Tracker.gg/dist/metadata.json [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+  "author": {
+    "name": "OGMatrix",
+    "id": "657290880947978241"
+  },
+  "service": "Tracker.gg",
+  "description": {
+    "en": "Tracker Network provides stats, global and regional leaderboards and much more to gamers around the world.",
+    "de": "Tracker Network bietet Spielern auf der ganzen Welt Statistiken, globale und regionale Bestenlisten und vieles mehr."
+  },
+  "url": ["tracker.gg", "fortnitetracker.com"],
```
[the remark this answers is not in the document]
Comment (OGMatrix) [reply websites/T/Tracker.gg/dist/metadata.json:12]: There are not totally different they are from the same author and the same company, just another website for extra tools.

Comment (ImRodry) [reply websites/T/Tracker.gg/dist/metadata.json:12]: The websites appear totally different and none of the code is shared, you should make two different presences

File: websites/T/Tracker.gg/presence.ts [hunk 14]
```diff
@@ -0,0 +1,524 @@
...
+      try {
+        const appName = document
+          .querySelector<HTMLDivElement>(
+            "#app > div.trn-wrapper > div.trn-container > div > main > div.apps > div.content-container > div > div > div > div > div.dashboard__analytics.card.bordered.responsive > header > div > div.left > div:nth-child(2) > div.title"
+          )
+          .textContent.replace("Dashboard", "");
+        presenceData.details = "Editing App";
+        presenceData.state = appName;
+        presenceData.smallImageKey = "logo";
+        presenceData.smallImageText = appName;
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:35]: You do not need to repeat the smallImageKey, it is already set in the state

File: websites/T/Tracker.gg/presence.ts [hunk 15]
```diff
@@ -0,0 +1,524 @@
...
+        presenceData.details = "Viewing:";
+        presenceData.state = "Developer Apps";
+      }
+    } else if (pathname.includes("/developers/docs")) {
+      try {
+        const title: string = document.querySelector(
+          "#app > div.trn-wrapper > div.trn-container > div > main > div.docs > div.content-container > div > div > h1"
+        ).textContent;
+        presenceData.details = "Viewing Docs";
+        presenceData.state = title;
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:49]: Copy this throughout your code, L51

```suggestion
        presenceData.details = "Viewing Docs";
        presenceData.state = document.querySelector<HTMLHeadingElement>(
          "#app > div.trn-wrapper > div.trn-container > div > main > div.docs > div.content-container > div > div > h1"
        ).textContent;
```

File: websites/T/Tracker.gg/presence.ts [hunk 16]
```diff
@@ -0,0 +1,524 @@
...
+        ).textContent,
+        image: string = document
+          .querySelector<HTMLImageElement>(
+            "#app > div.trn-wrapper > div.trn-container > div > main > div.content.no-card-margin > div.ph > div.ph__container > div.ph-avatar > svg > image"
+          )
+          .getAttribute("href");
+      presenceData.details = "Viewing Valorant Profile:";
+      presenceData.state = playerName + playerTag;
+      presenceData.smallImageKey = "valorant";
+      presenceData.smallImageText = playerName;
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:99]: Repetition

File: websites/T/Tracker.gg/presence.ts [hunk 17]
```diff
@@ -0,0 +1,524 @@
...
+          "#app > div.trn-wrapper > div.trn-container > div > main > div.content.no-card-margin > div.ph > div.ph__container > div.ph-details > div.ph-details__identifier > span > span.trn-ign__username"
+        ).textContent,
+        playerTag: string = document.querySelector(
+          "#app > div.trn-wrapper > div.trn-container > div > main > div.content.no-card-margin > div.ph > div.ph__container > div.ph-details > div.ph-details__identifier > span > span.trn-ign__discriminator"
+        ).textContent,
+        image: string = document
+          .querySelector<HTMLImageElement>(
+            "#app > div.trn-wrapper > div.trn-container > div > main > div.content.no-card-margin > div.ph > div.ph__container > div.ph-avatar > svg > image"
+          )
+          .getAttribute("href");
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:95]: Variables are only used once, you should rebase.

Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:95]: href might exist on HTMLImageElement

File: websites/T/Tracker.gg/presence.ts [hunk 18]
```diff
@@ -0,0 +1,524 @@
...
+        presenceData.smallImageText = "TRN";
+        presenceData.largeImageKey = "valorant";
+      } else {
+        try {
+          const guideName: string = document.querySelector(
+              "#app > div.trn-wrapper > div.trn-container > div > main > article > div.guide__header > h1"
+            ).textContent,
+            authorName: string = document.querySelector(
+              "#app > div.trn-wrapper > div.trn-container > div > main > article > div.guide__main > div > div.guide-main__hero.card.header-bordered.responsive > div.guide-main-metadata > div.guide-main__author.guide-main-metadata__author > span.guide-main__author-username"
+            ).textContent;
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/presence.ts:158]: Variables are only used once

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
check if your variables are only being used once, seems quite repetitve

File: websites/T/Tracker.gg/dist/metadata.json [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+    "id": "657290880947978241"
+  },
+  "service": "Tracker.gg",
+  "description": {
+    "en": "Tracker Network provides stats, global and regional leaderboards and much more to gamers around the world.",
+    "de": "Tracker Network bietet Spielern auf der ganzen Welt Statistiken, globale und regionale Bestenlisten und vieles mehr."
+  },
+  "url": ["tracker.gg", "fortnitetracker.com", "thetrackernetwork.com"],
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/YkjUiwX.png",
```
Comment (EncryptedDev) [inline websites/T/Tracker.gg/dist/metadata.json:14]: Not 512x512

File: websites/T/Tracker.gg/presence.ts (hunk 17 above)
Comment (OGMatrix) [reply websites/T/Tracker.gg/presence.ts:95]: Doesn't exist on HTMLImageElement

**Comment by OGMatrix** [comment]
How can I resolve the conflict?

**Comment by ImRodry** [comment]
Rebase your branch

**Comment by EncryptedDev** [comment]
you definitely didn't rebase properly

**Comment by OGMatrix** [comment]
> Rebase your branch

So delete the Repository and fork it again, right?

**Comment by ImRodry** [comment]
Wrong, google how to rebase or fix the merge conflicts here on the github website

**Comment by OGMatrix** [comment]
> Wrong, google how to rebase or fix the merge conflicts here on the github website

It doesn't work, it says everything is up to date.

**Comment by ImRodry** [comment]
I’m sure it doesn’t. Use the presence-dev channel on Discord if you need help

**Comment by OGMatrix** [comment]
> I’m sure it doesn’t. Use the presence-dev channel on Discord if you need help

What should I write in the Channel?

**Comment by OGMatrix** [comment]
Opened a new Pull Request for the Tracker Presence, this will be closed!
You can review it here: https://github.com/PreMiD/Presences/pull/5573

**Pull request closed without merging**